### PR TITLE
Generalized VT::point_gradients() for multiple components

### PR DESCRIPTION
**Pull request by peterrum**
FYI @jh66637

### CODE REVIEW

**Review by tamiko — APPROVED** [review]
Can we add a quick test for this?

**Review by kronbichler — APPROVED** [review]

**Review by peterrum — COMMENTED** [review]
@jh66637 Thanks for the test. The serial version of CI is complaining. You need to rename the test output to `vector_tools_evaluate_at_points_07.with_p4est.output` or alternatively switch to a serial triangulation, since your test is serial.

File: include/deal.II/numerics/vector_tools_evaluate.h [hunk 1]
```diff
@@ -343,6 +343,42 @@ namespace VectorTools
...
+          case EvaluationFlags::avg:
+            {
+              Tensor<1, n_components, Tensor<rank, dim, Number>> temp;
+
+              for (unsigned int j = 0; j < values.size(); ++j)
+                for (unsigned int i = 0; i < n_components; ++i)
+                  temp[i] = temp[i] + values[j][i];
+
+              for (unsigned int i = 0; i < n_components; ++i)
+                temp[i] /= (Number(1.0) * values.size());
```
Comment (kronbichler) [inline include/deal.II/numerics/vector_tools_evaluate.h:368]: Does it work to simply write
```suggestion
                temp[i] /= Number(values.size());
```

**Comment by jh66637** [comment]
Let me know if there is anything else that has to be done to get this merged :)

**Review by kronbichler — APPROVED** [review]

**Comment by kronbichler** [comment]
I see the test `remote_point_evaluation/vector_tools_evaluate_at_points_07` fail since a few days, see e.g. https://cdash.dealii.43-1.org/test/27512067 - can this PR be the culprit?

**Pull request merged**